tests/E2E/test/campaigns/full/15_design/3_link_widget/01_linkWidget.js

```diff
@@ -419,22 +418,37 @@ scenario('Create, edit, delete LinkWidget with different HOOK ', () => {
   commonLinkWidget.createWidget('EditHook', 'displayFooter');
 
   scenario('Edit the created linkwidget hook from "displayFooter" to "displayleftcolumn"', client => {
-    test('should go to "Design - Link Widget" page', () => client.goToSubtabMenuPage(Menu.Improve.Design.design_menu, Menu.Improve.Design.link_widget_submenu));
-    test('should click on Display Footer "EditHook" linkwidget edit button', () => client.waitForExistAndClick(LinkWidget.edit_display_footer_created_hook));
-    test('should choose the hook "displayLeftColumn"', () => client.waitAndSelectByVisibleText(LinkWidget.hook_select, "displayLeftColumn"));
-    test('should select All the "content pages"', () => client.waitForExistAndClick(LinkWidget.select_all_content_page));
-    test('should deactivate all product pages"', () => {
-      return promise
-        .then(() => client.waitForExistAndClick(LinkWidget.select_all_product_page))
-        .then(() => client.waitForExistAndClick(LinkWidget.select_all_product_page));
+    test('should go to "Design - Link Widget" page', async () => await client.goToSubtabMenuPage(Menu.Improve.Design.design_menu, Menu.Improve.Design.link_widget_submenu));
+    test('should click on Display Footer "EditHook" linkwidget edit button', async () => await client.waitForExistAndClickJs(LinkWidget.edit_display_footer_created_hook));
+    test('should choose the hook "displayLeftColumn"', async () => await client.waitAndSelectByVisibleText(LinkWidget.hook_select, "displayLeftColumn"));
+    test('should unselect All the "content pages"', async () => {
+      let i = 1;
+      do{
+        await client.isVisible(LinkWidget.select_all_content_page.replace('%POS',i));
+        if(global.isVisible)
+          client.scrollWaitForExistAndClick(LinkWidget.select_all_content_page.replace('%POS',i));
+        i++;
+      }while(global.isVisible);
+    });
+    test('should deactivate all product pages"', async () => {
+      let i = 1;
+      do{
+        await client.isVisible(LinkWidget.select_all_product_page.replace('%POS',i));
+        if(global.isVisible)
+          client.scrollWaitForExistAndClick(LinkWidget.select_all_product_page.replace('%POS',i));
+        i++;
+      }while(global.isVisible);
     });
-    test('should deactivate all "static content"', () => {
-      return promise
-        .then(() => client.waitForExistAndClick(LinkWidget.select_all_static_content))
-        .then(() => client.waitForExistAndClick(LinkWidget.select_all_static_content));
+    test('should deactivate all "static content"', async () => {
+      let i = 1;
+      do{
+        await client.isVisible(LinkWidget.select_all_static_content.replace('%POS',i));
+        if(global.isVisible)
+          client.scrollWaitForExistAndClick(LinkWidget.select_all_static_content.replace('%POS',i));
+        i++;
+      }while(global.isVisible);
```

Comment:
I see this content is a duplicate from `tests/E2E/test/campaigns/common_scenarios/linkwidget.js`. To make future changes easier to apply, could you please refacto your code to have a single method containing these lines and then called by your `linkwidget` scenarios?

Reply:
Thanks for suggestion, it's done